lib/ibm_cloud_sdk_core/api_exception.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require("json")
+# Custom exception class for errors returned from the APIs
+class ApiException < StandardError
+  attr_reader :code, :error, :info, :transaction_id, :global_transaction_id
+  # :param HTTP::Response response: The response object from the API
+  def initialize(code: nil, error: nil, info: nil, transaction_id: nil, global_transaction_id: nil, response: nil)
+    if code.nil? || error.nil?
+      @code = response.code
+      @error = response.reason
+      unless response.body.empty?
+        body_hash = JSON.parse(response.body.to_s)
+        @code = body_hash["code"] || body_hash["error_code"] || body_hash["status"]
+        @error = body_hash["error"] || body_hash["error_message"] || body_hash["statusInfo"] || body_hash["description"]
```

Comment:
Per issue 812 (https://github.ibm.com/arf/planning-sdk-squad/issues/818) we have inconsistencies in the field names that we try to retrieve from an error response.   We should scrub this list of fields to make sure we're complying with the API guidelines.   I think at a minimum we should add "message" to the list.
See this page for more details:  https://pages.github.ibm.com/CloudEngineering/api_handbook/design/errors.html

Reply:
updated.